pennylane_lightning_gpu/lightning_gpu.py

```diff
@@ -252,6 +252,8 @@ def __init__(
 
             super().__init__(wires, shots=shots, r_dtype=r_dtype, c_dtype=c_dtype)
 
+            self._dp = DevPool()
```

Comment:
Why was this moved? Is it needed for this PR?

Reply:
This line was moved here because `self_.dp` is needed for both single GPU backend and multi-node/multi-GPU backend, but was only kept at the multi-node version in the previous implementation. It was a bug that didn't show up in previous unit tests check. We need this change to pass the CI checks.

Reply:
I think I recall we moved this in the same way in the past to ensure it is single and multinode accessible. Maybe another merge overwrite this change. All good to keep it here.